src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/IdentifiersShouldHaveCorrectSuffix.cs

```diff
@@ -147,16 +147,22 @@ private static void AnalyzeCompilationStart(CompilationStartAnalysisContext cont
                 }
                 , SymbolKind.NamedType);
 
-                context.RegisterSymbolAction((saContext) =>
+                var eventArgsType = WellKnownTypes.EventArgs(context.Compilation);
+                if (eventArgsType != null)
                 {
-                    const string eventHandlerString = "EventHandler";
-                    var eventSymbol = saContext.Symbol as IEventSymbol;
-                    if (!eventSymbol.Type.Name.EndsWith(eventHandlerString, StringComparison.Ordinal))
+                    context.RegisterSymbolAction((saContext) =>
                     {
-                        saContext.ReportDiagnostic(eventSymbol.CreateDiagnostic(DefaultRule, eventSymbol.Type.Name, eventHandlerString));
-                    }
-                },
-                SymbolKind.Event);
+                        const string eventHandlerString = "EventHandler";
+                        var eventSymbol = (IEventSymbol)saContext.Symbol;
+                        if (!eventSymbol.Type.Name.EndsWith(eventHandlerString, StringComparison.Ordinal) &&
+                            eventSymbol.Type.TypeKind == TypeKind.Delegate &&
+                            ((INamedTypeSymbol)eventSymbol.Type).DelegateInvokeMethod?.HasEventHandlerSignature(eventArgsType) == true)
```

Comment:
don't you still need to check whether type is source symbol? otherwise, user still can't rename the symbol? also, since it shows squiggle on reference, it might show multiple squiggles on same delegate?

Reply:
It matches legacy FxCop behavior, but I agree this will reduce noise further. Will add a check to verify symbol is in source.

> it might show multiple squiggles for same delegate

I think that is fine, given delegate might be used at non-event locations as well, so fix has to be per event with that type.

Reply:
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/1980/commits/0df5197b78879c0cb4d52b11bff694ff75b20489